ui/src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -283,6 +283,45 @@ const FlexTopBar = (props: {toolbar: Toolbar | Observable<Toolbar>}) => {
     );
 };
 
+type AppSorter = (a: models.Application, b: models.Application) => number;
+
+const AppSorters: {[key: string]: AppSorter} = {
+    'Name': ((a, b) => {
+        if (a && a.metadata && b && b.metadata) {
+            return a.metadata.name < b.metadata.name ? -1 : 1;
+        }
+        return 1;
+    }) as AppSorter,
+    'Deployment Time': ((a, b) => (a.status.observedAt < b.status.observedAt ? 1 : -1)) as AppSorter
```

Comment:
Looks like the `ApplicationStatus.ObservedAt` field has been deprecated, probably `ApplicationStatus.OperationState.FinishedAt` would be a better replacement?